platform/core.windows/src/org/netbeans/core/windows/services/NbPresenter.java

```diff
@@ -1591,6 +1591,18 @@ private void initBounds() {
      */
     private Window findFocusedWindow() {
         Window w = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusedWindow();
+        if( w == null ) {
+            // PR#5280
+            LOG.fine( () -> "No focused window, find mainWindow" );
+            for( Frame f01 : Frame.getFrames() ) {
```

Comment:
Nitpick: call `findMainWindow`  ?

Reply:
> call `findMainWindow` ?

@sdedic I'd like to do that; can you give me some guidance? Need to find a non-API file in `platform` that's OK to access from both `core.windows` and `openide.util.ui` where `findMainWindow` can be declared as a public method. Such a file must exist, any pointers?

Reply:
Ah ;) my bad, I overlooked the module boundary. No suggestion that 'feels' right then. An option would be to add a API/SPI method on e.g. DialogDisplayer or WindowManager.

BTW this method uses `KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusedWindow()` while `getCurrentGraphicsConfiguration()`  (in diff below) uses `KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner()` then `getWindowAncestor()`.  Deliberate difference or inconsistency ?

Reply:
I've wondered about that myself; are the two equivalent?

I'm copying your comment to #4887 which are notes for a follow on PR addressing some stuff in `Utilities`. The opening comment of this PR mentions issues for the next PR.